progenitor-impl/src/method.rs

```diff
@@ -1134,6 +1134,11 @@ impl Generator {
                 (#hook)(&#client.inner, &#request_ident);
```

Comment:
let's just make this mut; it was probably a mistake to diverge the non-async and async variants. Let's not compound the mistake.